cmd/controller/app/app.go

```diff
@@ -296,48 +289,36 @@ func (a *App) runKubeServer(ctx context.Context, log *logging.Logger, client *ku
 }
 
 func (a *App) runMetricsHTTPServer(ctx context.Context, log *logging.Logger) error {
-	e := echo.New()
-	e.HideBanner = true
-	e.Debug = false
+	log.Info("running http server")
+	defer log.Info("stopping http server")
 
-	e.Use(middleware.Recover())
-	e.GET("/metrics", echo.WrapHandler(promhttp.Handler()))
-	e.GET("/debug/pprof/*item", echo.WrapHandler(http.DefaultServeMux))
+	mux := http.NewServeMux()
```

Comment:
Why did you switch from `echo` to golang built in http?

Reply:
On agent we already use simple std http. While setting up pyroscope I was not able to make it properly work with echo, something is off with the routing. Did not investigate much why but It doesn't make much sense to have echo just for this use case. Was easier to fix by switching to std as was done in the agent.